parts/windows/windowscalicofunc.ps1

```diff
@@ -27,10 +27,27 @@ function GetCalicoKubeConfig {
         [parameter(Mandatory=$false)] $KubeConfigPath = "c:\\k\\config"
     )
 
-    $name=c:\k\kubectl.exe --kubeconfig=$KubeConfigPath get secret -n $CalicoNamespace --field-selector=type=kubernetes.io/service-account-token --no-headers -o custom-columns=":metadata.name" | findstr $SecretName | select -first 1
+    # When creating Windows agent pools with the system Linux agent pool, the service account for calico may not be available in provisioning Windows agent nodes.
+    # So we need to wait here until the service account for calico is available
+    $name=""
+    $retryCount=0
+    $retryInterval=5
+    $maxRetryCount=120 # 10 minutes

```

Comment:
Have we checked apiserver connectivity before this loop?

Reply:
No. To simplify it in provisioning script, we will retry until we get the service account. It will not get the service account if it cannot connect to apiserver.

Reply:
Unfortunately, we don't have apiserver and mcr connectivity check as Linux. Something we need to improve. I will create a work item to track this.

Reply:
@feiskyer Updated pr to check API server connectivity before getting calico service account from API server